tests/_internal/mbed_tools/test_config_command.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+import pathlib
+from unittest import TestCase, mock
+
+from click.testing import CliRunner
+from tabulate import tabulate
+
+from mbed_build._internal.mbed_tools.config import (
+    config,
+    _build_tabular_output,
+    _build_json_output,
+    _build_legacy_output,
+)
+from tests._internal.config.factories import BuildConfigFactory, OptionFactory
+
+
+@mock.patch("mbed_build._internal.mbed_tools.config._build_legacy_output")
+@mock.patch("mbed_build._internal.mbed_tools.config._build_json_output")
+@mock.patch("mbed_build._internal.mbed_tools.config._build_tabular_output")
+@mock.patch("mbed_build._internal.mbed_tools.config.assemble_config")
+class TestConfig(TestCase):
+    mbed_target = "K64F"
+    project_path = "somewhere"
+
+    def test_config_table(self, assemble_config, build_tabular_output, build_json_output, build_legacy_output):
+        runner = CliRunner()
+        result = runner.invoke(config, ["-m", self.mbed_target, "-p", self.project_path])
+
+        self.assertEqual(result.exit_code, 0)
+        assemble_config.assert_called_once_with(self.mbed_target, pathlib.Path(self.project_path))
+        build_tabular_output.assert_called_once_with(assemble_config.return_value)
```

Comment:
I think it'd be better to avoid unnecessary mocking and do something along the lines of:

```
assemble_config.return_value = BuildConfigFactory()
self.assertEqual(result.output, build_tabular_output(assemble_config.return_value))
```

Is that possible?

Asserting other output mocks have not been called does not really provide value here. 🤔

Reply:
Yeah thats true and yay for less mocking

Reply:
a6771e3